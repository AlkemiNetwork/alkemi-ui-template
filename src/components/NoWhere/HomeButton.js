import React from "react";
import PropTypes from "prop-types";

import Button from "@material-ui/core/Button";
import { withStyles } from "@material-ui/core/styles";

const styles = theme => ({
    button: {
        margin: theme.spacing.unit
    }
});

const NoWhereHomeButton = ({ classes, onClick }) => (
    <Button
        variant="raised"
        color="default"
        className={classes.button}
        onClick={onClick}
    >
        Bring me back home!
    </Button>
);

NoWhereHomeButton.propTypes = {
    classes: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired
};

export default withStyles(styles)(NoWhereHomeButton);
