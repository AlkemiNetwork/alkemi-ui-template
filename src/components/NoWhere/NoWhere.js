import React from "react";
import PropTypes from "prop-types";

import ReportIcon from "@material-ui/icons/Report";
import { withStyles } from "@material-ui/core/styles";
import { grey } from "@material-ui/core/colors";

import NoWhereHomeButton from "../../containers/NoWhere/HomeButton";

const styles = theme => ({
    nowhere: {
        textAlign: "center"
    },
    nowhereHeader: {
        padding: theme.spacing.unit * 3
    },
    nowhereTitle: {
        fontSize: "1.5em",
        color: grey[500]
    },
    nowhereIcon: {
        height: 80,
        width: 80,
        color: grey[500]
    }
});

const NoWhere = ({ classes }) => (
    <div className={classes.nowhere}>
        <div className={classes.nowhereHeader}>
            <ReportIcon className={classes.nowhereIcon} />
            <h1 className={classes.nowhereTitle}>
                Oups... It seems that you are lost!
            </h1>
        </div>
        <NoWhereHomeButton />
    </div>
);

NoWhere.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(NoWhere);
