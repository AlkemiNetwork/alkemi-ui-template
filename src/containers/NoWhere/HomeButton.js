import { connect } from "react-redux";
import { push } from "connected-react-router";

import NoWhereHomeButton from "../../components/NoWhere/HomeButton";
import { HOME } from "../../constants/routes";

const mapDispatchToProps = dispatch => ({
    onClick: () => dispatch(push(HOME))
});

export default connect(
    undefined,
    mapDispatchToProps
)(NoWhereHomeButton);
