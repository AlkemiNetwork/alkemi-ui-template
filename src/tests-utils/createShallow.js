import { shallow as _shallow } from "enzyme";

const createShallow = ({ dive }) => {
    const shallow = (node, options) => {
        let wrapper = _shallow(node, options);

        for (let i = 0; i < dive; i++) {
            if (wrapper.name().match(/^WithWidth(.+)$/)) {
                // Diving through a Meterial-UI WidthWith component can not be performed
                // by default because of React.Fragment used in withWidth
                // It is a known issue: https://github.com/airbnb/enzyme/issues/1213
                wrapper = wrapper
                    .dive()
                    .children()
                    .at(0);
            } else {
                wrapper = wrapper.dive();
            }
        }
        return wrapper;
    };

    return shallow;
};

export default createShallow;
