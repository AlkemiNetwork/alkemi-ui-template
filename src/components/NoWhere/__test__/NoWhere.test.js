import React from "react";
import toJson from "enzyme-to-json";

import ReportIcon from "@material-ui/icons/Report";

import NoWhere from "../NoWhere";
import NoWhereHomeButton from "../../../containers/NoWhere/HomeButton";
import createShallow from "../../../tests-utils/createShallow";

describe("<NoWhere />", () => {
    let shallow;
    let wrapper;

    beforeAll(() => {
        shallow = createShallow({ dive: 1 });
    });

    beforeEach(() => {
        wrapper = shallow(<NoWhere />);
    });

    it("renders 1 ReportIcon", () => {
        expect(wrapper.find(ReportIcon).length).toEqual(1);
    });

    it("renders 1 BackHomeButton", () => {
        expect(wrapper.find(NoWhereHomeButton).length).toEqual(1);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
