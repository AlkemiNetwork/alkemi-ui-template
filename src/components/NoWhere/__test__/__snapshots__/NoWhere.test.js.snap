// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoWhere /> matches snapshot 1`] = `
<div
  className="NoWhere-nowhere-1"
>
  <div
    className="NoWhere-nowhereHeader-2"
  >
    <pure(ReportIcon)
      className="NoWhere-nowhereIcon-4"
    />
    <h1
      className="NoWhere-nowhereTitle-3"
    >
      Oups... It seems that you are lost!
    </h1>
  </div>
  <Connect(WithStyles(NoWhereHomeButton)) />
</div>
`;
